docs/xkcd_extension_tutorial.md

```diff
@@ -0,0 +1,653 @@
+# Let's Make an xkcd JupyterLab Extension
+
```

Comment:
This document is rather long.  Could you consider including a table of contents with section links?  Or perhaps breaking up the document?

Reply:
I totally agree. I was thinking the one in the left sidebar of the alabaster theme is a good start since it links to all of the section headings.

![screen shot 2017-09-18 at 2 58 47 pm](https://user-images.githubusercontent.com/153745/30559331-46a5a688-9c82-11e7-8e05-dec07e918a86.png)

It does scroll with the page, however, so it's still somewhat annoying to jump around. I'm open to other ideas.

Reply:
This would work if sections were grouped into higher level groupings eg `Installation`  `Initialize Extension With The Cookiecutter`

Reply:
Sure. I'll take a pass at them on the next update.

Reply:
How about these, where only the # and ## level headings appears in the table of contents:

```
# Let's Make an xkcd JupyterLab Extension

## Setup a development environment
### Install conda using miniconda
### Install NodeJS, JupyterLab, etc. in a conda environment

## Create an extension project
### Initialize the project from a cookiecutter
### Build and link the extension for development
### See the initial extension in action
### Commit what you have to git

## Add an xkcd widget
### Show an empty panel
### Show a comic in the panel

## Improve the widget behavior
### Center the comic and add attribution
### Show a new comic on demand
### Restore panel state when the browser refreshes

## Publish your extension to npmjs.org

## Learn more
```

Reply:
LGTM